src/snapshot/tree.rs

```diff
@@ -174,6 +207,10 @@ impl RojoTree {
     fn remove_metadata(&mut self, id: Ref) {
         let metadata = self.metadata_map.remove(&id).unwrap();
 
+        if let Some(specified) = metadata.specified_id {
+            self.specified_id_to_refs.remove(&specified);
+        }
+
```

Comment:
This probably needs a duplicate ID check as well?

Reply:
On second thought.... no, that doesn't make sense 🤪

Reply:
On third thought (sorry for the wavering, this was a last-minute addition)...

If I'm thinking about this right, this *could* cause an inappropriate removal from `specified_id_to_refs` when the ID is a duplicate. We should check that the entry in the map actually corresponds to the instance we're removing metadata from.

Reply:
You're correct, this could cause some problems. We need to uphold the invariant that there can't be duplicates and this doesn't at the moment. I'll fix that.